environment/config.go

```diff
@@ -95,112 +127,131 @@ type ConfigureResults struct {
 
 // ConfigurationParameters are options used to create new ConfigureResults
 type ConfigurationParameters struct {
-	InstrumentOpts instrument.Options
-	HashingSeed    uint32
-	HostID         string
+	InstrumentOpts   instrument.Options
+	HashingSeed      uint32
+	HostID           string
+	NamespaceTimeout time.Duration
 }
 
 // Configure creates a new ConfigureResults
 func (c Configuration) Configure(cfgParams ConfigurationParameters) (ConfigureResults, error) {
+	var emptyConfig ConfigureResults
+
+	if c.Service.SDConfig.InitTimeout == 0 {
+		c.Service.SDConfig.InitTimeout = defaultSDTimeout
+	}
 
+	configSvcClientOpts := c.Service.NewOptions().
+		SetInstrumentOptions(cfgParams.InstrumentOpts).
+		SetServiceDiscoveryConfig(c.Service.SDConfig)
+	configSvcClient, err := etcdclient.NewConfigServiceClient(configSvcClientOpts)
+	if err != nil {
+		err = fmt.Errorf("could not create m3cluster client: %v", err)
+		return emptyConfig, err
+	}
+
+	if c.Service != nil {
+		return c.newDynamicConfigRes(configSvcClient, cfgParams)
+	}
+
+	if c.Static != nil {
```

Comment:
Should probably check that they are not both set too?  Should be invalid to specify both.